### PR TITLE
feat: allow user to specify APISIX id in config.yaml

### PR DESCRIPTION
**Pull request by tzssangglass**
### What this PR does / why we need it:

#2810 
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: FAQ_CN.md [hunk 1]
```diff
@@ -250,3 +250,14 @@ APISIX 主要使用 [etcd.watchdir](https://github.com/api7/lua-resty-etcd/blob/
 * 如果监听目录有数据更新：etcd 将立刻返回订阅(毫秒级)到的新数据，APISIX 将它更新到内存缓存。
 
 借助 etcd 增量通知毫秒级特性，APISIX 也就完成了毫秒级的配置同步。
+
+## 如何自定义 APISIX 实例 id
+
+默认情况下，APISIX 使用 `uuid` 作为实例 id，存储在`/conf/apisix.uid`中，在 APISIX 第一次启动时生成，不会改变。
```
Comment (spacewander) [inline FAQ_CN.md:256]: The subject of this sentence is inconsistent.
What about:
APISIX 会从 `conf/apisix.uid` 中读取实例 id。
如果找不到，且没有配置 id，APISIX 会生成一个 `uuid` 作为实例 id。

File: FAQ_CN.md [hunk 2]
```diff
@@ -250,3 +250,14 @@ APISIX 主要使用 [etcd.watchdir](https://github.com/api7/lua-resty-etcd/blob/
...
+
+如果你想指定一个有意义的 id 来绑定 APISIX 实例与你的内部系统，你可以在`conf/config.yaml`中进行配置，示例：
```
Comment (spacewander) [inline FAQ_CN.md:258]: "与" should be "到"

File: apisix/core/id.lua [hunk 3]
```diff
@@ -65,9 +66,15 @@ function _M.init()
         return
     end
 
-    uuid.seed()
-    apisix_uid = uuid.generate_v4()
-    log.notice("not found apisix uid, generate a new one: ", apisix_uid)
+    --allow user to specify a meaningful id as apisix instance id
+    local local_conf = fetch_local_conf()
+    if local_conf and local_conf.apisix and local_conf.apisix.id then
```
Comment (spacewander) [inline apisix/core/id.lua:71]: Can we use try_read_attr?
https://github.com/apache/apisix/blob/ce1d40b42f056f46b20386e1937fc81ba69e84c6/apisix/core/table.lua#L63

**Review by spacewander — CHANGES_REQUESTED** [review]

File: FAQ_CN.md (hunk 2 above)
Comment (tzssangglass) [reply FAQ_CN.md]: solve it

File: FAQ_CN.md (hunk 1 above)
Comment (tzssangglass) [reply FAQ_CN.md]: solve it

File: apisix/core/id.lua (hunk 3 above)
Comment (tzssangglass) [reply apisix/core/id.lua]: solve it

File: FAQ_CN.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tzssangglass) [reply FAQ_CN.md]: solve it

**Review by spacewander — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
@tzssangglass many thx

**Pull request merged**